contrib/kind.sh

```diff
@@ -593,29 +597,72 @@ set_cluster_cidr_ip_families() {
   fi
 }
 
+create_local_registry() {
+    # create registry container unless it already exists
+    if [ "$($OCI_BIN inspect -f '{{.State.Running}}' "${KIND_LOCAL_REGISTRY_NAME}" 2>/dev/null || true)" != 'true' ]; then
+      $OCI_BIN run \
+        -d --restart=always -p "127.0.0.1:${KIND_LOCAL_REGISTRY_PORT}:5000" --name "${$KIND_LOCAL_REGISTRY_NAME}" \
+        registry:2
+    fi
+}
+
+connect_local_registry() {
+    # connect the registry to the cluster network if not already connected
+    if [ "$($OCI_BIN inspect -f='{{json .NetworkSettings.Networks.kind}}' "${KIND_LOCAL_REGISTRY_NAME}")" = 'null' ]; then
+      $OCI_BIN network connect "kind" "${KIND_LOCAL_REGISTRY_NAME}"
+    fi
+
+    # Document the local registry
+    # https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+data:
+  localRegistryHosting.v1: |
+    host: "localhost:${KIND_LOCAL_REGISTRY_PORT}"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+EOF
+
+}
+
 create_kind_cluster() {
   # Output of the j2 command
   KIND_CONFIG_LCL=${DIR}/kind-${KIND_CLUSTER_NAME}.yaml
 
-    ovn_ip_family=${IP_FAMILY} \
-    ovn_ha=${OVN_HA} \
-    net_cidr=${NET_CIDR} \
-    svc_cidr=${SVC_CIDR} \
-    use_local_registy=${KIND_LOCAL_REGISTRY} \
-    dns_domain=${KIND_DNS_DOMAIN} \
-    ovn_num_master=${KIND_NUM_MASTER} \
-    ovn_num_worker=${KIND_NUM_WORKER} \
-    cluster_log_level=${KIND_CLUSTER_LOGLEVEL:-4} \
-    j2 "${KIND_CONFIG}" -o "${KIND_CONFIG_LCL}"
+  ovn_ip_family=${IP_FAMILY} \
+  ovn_ha=${OVN_HA} \
+  net_cidr=${NET_CIDR} \
+  svc_cidr=${SVC_CIDR} \
+  use_local_registy=${KIND_LOCAL_REGISTRY} \
+  dns_domain=${KIND_DNS_DOMAIN} \
+  ovn_num_master=${KIND_NUM_MASTER} \
+  ovn_num_worker=${KIND_NUM_WORKER} \
+  cluster_log_level=${KIND_CLUSTER_LOGLEVEL:-4} \
+  j2 "${KIND_CONFIG}" -o "${KIND_CONFIG_LCL}"
 
   # Create KIND cluster. For additional debug, add '--verbosity <int>': 0 None .. 3 Debug
   if kind get clusters | grep ovn; then
     delete
   fi
+  
+  if [[ "${KIND_LOCAL_REGISTRY}" == true ]]; then
```

Comment:
So hopefully re-using the existing flag `--local-kind-registry` with a slightly different meaning does not disrupt anyone's workflow but that's a thought I had.

Reply:
If the local registry is already there is not started, I think that was how it was working before.